src/classes/RD_RecurringDonations_TEST2.cls

```diff
@@ -1084,36 +1091,48 @@ private with sharing class RD_RecurringDonations_TEST2 {
         test.startTest();
         insert rdlist;
         test.stopTest();
-        
+
         list<Opportunity> listOpp28th = [select Id, CloseDate from Opportunity where npe03__Recurring_Donation__r.Name = 'test28th' order by CloseDate];
-        system.assertEquals(12, listOpp28th.size());
+        system.assertEquals(12, listOpp28th.size(), 'There should be 12 Opps with a Close Date of the 28th');
         for (Opportunity opp : listOpp28th) {
-            system.assertEquals(28, opp.CloseDate.Day());
+            system.assertEquals(28, opp.CloseDate.Day(), 'The Opp.CloseDate.Day should be the 28th');
         }
 
```

Comment:
@force2b I think that it might be worth splitting this into 2 separate tests. One for the 28th and the other for the 30th.